### ISSUE TITLE
fit() validation processing should also be batched

### ISSUE THREAD

**Issue by elliottd** [fchollet/keras#384]
The current implementation of calculating validation loss tries to load the entire validation dataset at once, which can easily cause OutOfMemory errors on large datasets in networks and/or in networks with a large number of parameters.

**Comment by fchollet** [comment]
This is fixed.
